source/js/ellipsis-util.js

```diff
@@ -0,0 +1,99 @@
+var ellipsis = require("html-ellipsis");
+var characterPerLine = require("./characters-per-line");
+
+var EllipsisUtil = {
+  /***
+   * @param {Object} lineStyles: Node with DOM elements that has a string
+   * @return {Boolean}
+   */
+  hasSpecificHeight: function(lineStyles) {
+    return lineStyles.height != "auto" && parseInt(lineStyles.height) != 0;
+  },
+
+  /***  
+   * @param {Node} element Node element.
+   * @return {Integer} Returns a font size.
+  * */
+  getFontSize: function(element) {
+    return parseInt(window.getComputedStyle(element).getPropertyValue("font-size"));
+  },
+
+  /***
+   * @summary Returns the width of the object 
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} width
+   */
+  getWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+  
+    var lineStyles = window.getComputedStyle(textLine);
+
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      return parseInt(lineStyles.width);
+    }
+  },
+
+  /***
+   * @summary Returns the width of the object's parent 
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} parent's width
+   */
+  getParentsWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+    
+    var lineStyles = window.getComputedStyle(textLine);
+          
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      var parentElement = textLine.parentElement;
+      var parentStyles = window.getComputedStyle(parentElement);
+
+      if(parentElement.className !== "multi-teaser-hotel-name") {
+        return parseInt(parentStyles.width);
+      }
+
+      var hotelRating = parentElement.querySelector(".mot_stars");
+      var hotelRatingWidth = window.getComputedStyle(hotelRating).width;
+
+      return parseInt(parentStyles.width) - parseInt(hotelRatingWidth) - parseInt(lineStyles.marginRight);
+    }
+  },
+};
+
+/*** 
+ * @summary Creates a EllipseableObject and pushes it into teaser. 
+ * @param {Node} element Node element.
+ * @param {Integer} numberOfLines One of the model's attributes.
+ * @param {Function} getStyles A self-made method to extract computed Styles.
+ */
+var EllipseableObject = function(element, getStylesFunction, numberOfLines){
+  this.element = element;
+  this.html = this.element.innerHTML;
+  this.fontSize = EllipsisUtil.getFontSize(this.element);
+  this.numberOfLines = numberOfLines;
+  this.getWidth = getStylesFunction;
```

Comment:
what is i now. a variable or a function? In the usage of this object it return an integer. And I would suggest another name for a variable/property than "getWidth", maybe "widthCallback" or so

Reply:
When you actually want to call the function, do you do it with Object.getWidth() or Object.widthCallback() ?

Renamed
getStylesFunction => getWidthCallback.
To meet the readability.

Technically we are storing a lot of anonymous functions inside variables - they are called function expressions which we are storing inside variables and call them, when they are needed. But maybe we don't want to call them Callback or Expression, because of that.

Technically, we are talking about closures.
"In JavaScript, functions are first-class objects, because they can have properties and methods just like any other object."
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions
"In programming languages, a closure, also lexical closure or function closure, is a technique for implementing lexically scoped name binding in a language with first-class functions."
https://en.wikipedia.org/wiki/Closure_(computer_programming)
And in the examples, you can see, that the stored functions are not necessarily called fooBarClosure
https://en.wikipedia.org/wiki/Closure_(computer_programming)#Example_1:_Reference_to_an_unbound_variable

Check it out in line 94, where it is being called.

Reply:
As you should know as developer, there is no force to us the method you want to use public the same as the internal used variable.

Object.getWidth() can be implement like:
getWith() {
 return this.fooBar;
}

Reply:
Maybe I was not clear of what I asked for. The variable or property name "getWith" is what I found strange, the parameter "getStylesFunction" was OK, beside that we give an integer inside, when I see it right in the usage in source/js/ellipsis-config.js:42 and source/js/ellipsis-config.js:43 at least from all of the description/docBlock and namings.

Reply:
(Meeting at around 5pm including Erik, Birgit and Aurelie)

Reply:
The team decided in a meeting at around 10AM 100%-ly without any exception to move on.
People involved: Evandro, Vitor, Ankush, Fillipe, Birgit
Moderator: Julian
Erik has decided not to participate at the voting.